### ISSUE TITLE
Question about discriminator regularization loss

### ISSUE THREAD

**Issue by pnsuau** [rosinality/stylegan2-pytorch#76]
Hello !

What's the purpose of `+ 0 * real_pred[0]` in the line 190 of train.py `(args.r1 / 2 * r1_loss * args.d_reg_every + 0 * real_pred[0]).backward()` ? 

My intuition is that you wanted to propagate the loss through the discriminator but in my context it doesn't work, gradients are zeros in the discriminator.

I think that the following line could fix it : `(args.r1 / 2 * r1_loss * args.d_reg_every * real_pred[0]/real_pred[0].detach()).backward()`

Thank you !

**Comment by rosinality** [comment]
Actually `0 * real_pred[0]` is not for make gradients. It is to trigger backward hooks of DistributedDataParallel to makes allgather gradients. backward of losses that computed by `autograd.grad` will not trigger this hooks, so I have used backward that uses model outputs but have zero gradients.
